[lldb][test][win][x86_64] Fix XFAIL and XPASS on LLDB API tests

kendalharland opened the pull request:
I'm currently working on getting the LLDB test suites to pass on Windows x86_64 which is not currently included in LLVM CI. These tests are currently failing in this configuration.

See https://github.com/llvm/llvm-project/issues/100474

See also https://github.com/llvm/llvm-project/issues/75936

JDevlieghere commented:
Rather than a regex for the triple, can you use a combination of `oslist` and `archs`? `expectedFailureAll` is supposed to combine the conditions, so the regex should be equivalent to: 

```
@expectedFailureAll(oslist=["windows"], archs=["x86_64"])
```

kendalharland commented:
> Rather than a regex for the triple, can you use a combination of `oslist` and `archs`? `expectedFailureAll` is supposed to combine the conditions, so the regex should be equivalent to:
> 
> ```
> @expectedFailureAll(oslist=["windows"], archs=["x86_64"])
> ```

Yes. I will fix this in each of the PRs I recently mailed you.

kendalharland commented:
> > Rather than a regex for the triple, can you use a combination of `oslist` and `archs`? `expectedFailureAll` is supposed to combine the conditions, so the regex should be equivalent to:
> > ```
> > @expectedFailureAll(oslist=["windows"], archs=["x86_64"])
> > ```
> 
> Yes. I will fix this in each of the PRs I recently mailed you.
> 
> Will fix the dark formatter issue above too.

Done

labath commented on lldb/test/API/commands/settings/use_source_cache/TestUseSourceCache.py:
```diff
@@ -18,7 +18,7 @@ def test_set_use_source_cache_false(self):
 
     @skipIf(hostoslist=no_match(["windows"]))
-    @skipIf(oslist=["windows"])  # Fails on windows 11
+    @expectedFailureAll(oslist=["windows"])
```
Looking at the tested functionality (and the surrounding code), this should probably be `hostoslist=windows`

kendalharland replied:
Re: @DavidSpickett comment below. I am using the lldb-aarch64-windows builder as an example:

- Run in cmd.exe
- LLDB_TEST_COMPILER is the newly built clang-cl (top-level build uses host's MSVC)
- `--skip-test-category=watchpoint`
- LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS=ON
- Python 3.9

My host system:
- Win 11 Home Version 10.0.22631 Build 22631
- Dell XPS 8960

I haven't explicitly disabled hardware code breakpoints. What is the default and how do I do that?

kendalharland replied:
To reproduce my local build if you have access to a machine and a clean checkout llvm/llvm-project: Run this in a developer command prompt in the root of the checkout:

```
set MSVC_INSTALL_DIR=C:\Program Files\Microsoft Visual Studio\2022\Community
set "PATH=%MSVC_INSTALL_DIR%\DIA SDK\bin\amd64;%PATH%"

cmake -G Ninja ^
      -S llvm ^
      -B build ^
      -D Python3_EXECUTABLE="C:\Program Files\Python312\python.exe" ^
      -D CMAKE_C_COMPILER="cl" ^
      -D CMAKE_CXX_COMPILER="cl" ^
      -D CMAKE_C_COMPILER_LAUNCHER=ccache ^
      -D CMAKE_CXX_COMPILER_LAUNCHER=ccache ^
      -D CMAKE_BUILD_TYPE="Release" ^
      -D CMAKE_LINKER="C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\Llvm\x64\bin\lld-link.exe" ^
      -D LLVM_ENABLE_PROJECTS="clang;lldb;lld" ^
      -D LLVM_HOST_TRIPLE="x86_64-unknown-windows-msvc" ^
      -D LLVM_ENABLE_LLD="YES" ^
      -D LLVM_ENABLE_LIBEDIT="NO" ^
      -D LLVM_ENABLE_DIA_SDK="OFF" ^
      -D LLDB_ENABLE_PYTHON=1 ^
      -D LLDB_PYTHON_EXE_RELATIVE_PATH="python.exe" ^
      -D LLDB_TEST_USER_ARGS=--skip-category=watchpoint ^
      -D LLDB_TEST_COMPILER="clang-cl" ^
      -D LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS=1 ^
      -D LLDB_USE_LLDB_SERVER=1 ^
      -D CMAKE_CXX_STANDARD=14
```

kendalharland replied:
I made the following changes and I'm seeing far fewer failing tests now:
- `LLDB_TEST_COMPILER` - leave this unset
- `CMAKE_C_COMPILER="C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\Llvm\x64\bin\clang-cl.exe"`
- `CMAKE_CXX_COMPILER="C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\Llvm\x64\bin\clang-cl.exe"`

In short the compilers I was using caused many of the tests to fail.  I am only seeing around 17 failures now in a clean checkout and will fix this one up. Thanks for the course correction!

labath reviewed (commented):
I'm somewhat surprised to see this many x86 xfails, given that we already have an arm64 windows bot, and that none of the tests I see here are very architecture-specific.

I wonder if this could be due to how you've configured your build. @DavidSpickett, do you have any insight here?

DavidSpickett commented:
> I wonder if this could be due to how you've configured your build. @DavidSpickett, do you have any insight here?

Notable things about the Windows on Arm buildbot:
* Builds using clang-cl
* Runs in cmd.exe
* Watchpoint testing is disabled, hardware code breakpoints too

I'd show a full test log but for whatever reason, it's not being verbose despite being asked to, working on that.

Commit:
[lldb][test][win][x86_64] XFAIL already failing API tests

These test are passing locally on win x86_64 using the following
configuration (some options ellided for brevity)

CMAKE_C_COMPILER     clang-cl.exe
CMAKE_CXX_COMPILER   clang-cl.exe
CMAKE_BUILD_TYPE     Release
CMAKE_LINKER         lld-link.exe
LLVM_HOST_TRIPLE="x86_64-unknown-windows-msvc"
LLDB_TEST_USER_ARGS=--skip-category=watchpoint
LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS=1

kendalharland commented:
Thanks for the assistance @labath and @DavidSpickett. After fixing my cmake configuration I'm actually seeing a few tests unexpectedly pass with different options:

    CMAKE_C_COMPILER     clang-cl.exe
    CMAKE_CXX_COMPILER   clang-cl.exe
    CMAKE_BUILD_TYPE     Release
    CMAKE_LINKER         lld-link.exe
    LLVM_HOST_TRIPLE     x86_64-unknown-windows-msvc
    LLDB_TEST_USER_ARGS  --skip-category=watchpoint
    LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS 1

Updated in the latest commit

kendalharland commented on lldb/test/API/python_api/thread/TestThreadAPI.py:
```diff
@@ -52,6 +52,7 @@ def test_negative_indexing(self):
         self.build()
         self.validate_negative_indexing()
  
+    @expectedFailureAll(oslist=["windows"], archs=["x86_64"])
```
I am still seeing this test fail with:
```
======================================================================
FAIL: test_StepInstruction_dwarf (TestThreadAPI.ThreadAPITestCase.test_StepInstruction_dwarf)
    Test that StepInstruction preserves the plan stack.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 1756, in test_method
    return attrvalue(self)
           ^^^^^^^^^^^^^^^
  File "C:\workspace\llvm-project\lldb\test\API\python_api\thread\TestThreadAPI.py", line 58, in test_StepInstruction
    self.step_instruction_in_called_function()
  File "C:\workspace\llvm-project\lldb\test\API\python_api\thread\TestThreadAPI.py", line 334, in step_instruction_in_called_function
    self.assertRegex(
AssertionError: Regex didn't match: '.*call_me\\(bool\\)$' not found in 'main' : Stopped in call_me(bool)
Config=x86_64-C:\workspace\llvm-project\build\bin\clang.exe
----------------------------------------------------------------------

DavidSpickett replied:
I recently updated this for Windows on Arm https://github.com/llvm/llvm-project/commit/3e06392c7db0eacfca94a176d430d9988b3ffbd6, that's where that regex came from.

We need to know if it's a cosmetic difference or whether it's stopped at a completely different place. Running the test with `lldb-dotest` and the `-t` flag will hopefully show you what it was doing.

kendalharland replied:
Thanks! I'll try that and update here.

kendalharland replied:
Here's the full output with 3e06392 cherry-picked. https://pastebin.com/LQ9QReBr

kendalharland replied:
If I change this line:

```
self.runCmd("expr -i 0 -- call_me(true)", check=False)
```
to this:

```
self.runCmd("expr -i 0 -- call_me(true)", check=True)
```

I see this error:

```
======================================================================
FAIL: test_StepInstruction_dwarf (TestThreadAPI.ThreadAPITestCase.test_StepInstruction_dwarf)
   Test that StepInstruction preserves the plan stack.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 1761, in test_method
    return attrvalue(self)
           ^^^^^^^^^^^^^^^
  File "C:\workspace\llvm-project\lldb\test\API\python_api\thread\TestThreadAPI.py", line 16, in test_StepInstruction
    self.step_instruction_in_called_function()
  File "C:\workspace\llvm-project\lldb\test\API\python_api\thread\TestThreadAPI.py", line 291, in step_instruction_in_called_function
    self.runCmd("expr -i 0 -- call_me(true)", check=True)
  File "C:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 1006, in runCmd
    self.assertTrue(self.res.Succeeded(), msg if (msg) else CMD_MSG(cmd))
AssertionError: False is not true : Command 'expr -i 0 -- call_me(true)
Error output:
error: Couldn't look up symbols:
  void call_me(bool)
Hint: The expression tried to call a function that is not present in the target, perhaps because it was optimized out by the compiler.
' did not return successfully
Config=x86_64-C:\workspace\llvm-project\build\bin\clang.exe
----------------------------------------------------------------------

kendalharland replied:
Seems like my Clang invocation is generating different debug information than expected:

```
PS C:\workspace\llvm-project> llvm-pdbutil.exe pretty -all C:\workspace\llvm-project\build\lldb-test-build.noindex\python_api\thread\TestThreadAPI.test_StepInstruction_dwarf\a.pdb | Select-String -Pattern call_me

  func [0x00001000+ 0 - 0x00001025-37 | sizeof= 37] (FPO) void __cdecl call_me(bool should_spin)
  func [0x00001000+ 0 - 0x00001025-37 | sizeof= 37] (FPO) void __cdecl call_me(bool should_spin)
  public [0x00001000] ?call_me@@YAX_N@Z
```

Is the issue here that debug symbols have the additional `should_spin` parameter name, thus LLDB is not able to look them up since it is search for `void call_me(bool)`?

dzhidzhoev replied:
I see the same function name/signature.
Btw, I'd recommend adding `--env;LLDB_USE_LLDB_SERVER=1` to `LLDB_TEST_USER_ARGS` cmake argument. Without it, lldb can't even stop on a breakpoint on my machine. It may be helpful if you haven't set it yet.

DavidSpickett commented:
> I'd show a full test log but for whatever reason, it's not being verbose despite being asked to, working on that.

I've fixed this, you can get a full test list by downloading the log file e.g. https://lab.llvm.org/buildbot/#/builders/141/builds/1139/steps/6/logs/stdio.

labath reviewed (commented):
This looks much better, and it looks like @dzhidzhoev is just about to remove some of the same XFAILS as well (in #100628). I suggest you two collaborate on the remaining ones.

dzhidzhoev commented on lldb/test/API/functionalities/multiple-slides/TestMultipleSlides.py:
```diff
@@ -12,6 +12,7 @@
 class MultipleSlidesTestCase(TestBase):
     NO_DEBUG_INFO_TESTCASE = True
 
+    @expectedFailureAll(oslist=["windows"], archs=["x86_64"])
```
Could you provide a failure log for this? It passes on our CI (Windows native x86_64).

```
********************
PASS: lldb-api :: functionalities/multiple-slides/TestMultipleSlides.py (530 of 1191)
Script:
--
C:/Python312/python.exe C:/lldb\test\API\dotest.py -u CXXFLAGS -u CFLAGS --env OBJCOPY=C:/build-lldb-native/./bin/llvm-objcopy.exe --env STRIP=C:/build-lldb-native/./bin/llvm-strip.exe --env LLVM_LIBS_DIR=C:/build-lldb-native/./lib --env LLVM_INCLUDE_DIR=C:/build-lldb-native/include --env LLVM_TOOLS_DIR=C:/build-lldb-native/./bin --arch x86_64 --build-dir C:/build-lldb-native/lldb-test-build.noindex --lldb-module-cache-dir C:/build-lldb-native/lldb-test-build.noindex/module-cache-lldb\lldb-api --clang-module-cache-dir C:/build-lldb-native/lldb-test-build.noindex/module-cache-clang\lldb-api --executable C:/build-lldb-native/./bin/lldb.exe --make c:/gl/b1/3rdpaty/bin/make-wfix.exe --compiler C:/build-lldb-native/./bin/clang.exe --dsymutil C:/build-lldb-native/./bin/dsymutil.exe --llvm-tools-dir C:/build-lldb-native/./bin --lldb-obj-root C:/build-lldb-native/tools/lldb --lldb-libs-dir C:/build-lldb-native/./lib --skip-category=watchpoint C:\lldb\test\API\functionalities\multiple-slides -p TestMultipleSlides.py
--
Exit Code: 0

Command Output (stdout):
--
lldb version 20.0.0git (https://gitlab-ci-token:glcbt-64_rzNTyT5_UDR5sLkLB8b6@gitlab.ninolab.accesssoftek.com/accesssoftek/lldb-test-scripts.git revision cbd39658b6aed4dbb66a2b775096974c9087e93d)
  clang revision cbd39658b6aed4dbb66a2b775096974c9087e93d
  llvm revision cbd39658b6aed4dbb66a2b775096974c9087e93d
Skipping the following test categories: ['watchpoint', 'libc++', 'libstdcxx', 'dwo', 'dsym', 'gmodules', 'debugserver', 'objc', 'fork', 'pexpect']


--
Command Output (stderr):
--
PASS: LLDB (C:\build-lldb-native\bin\clang.exe-x86_64) :: test_mulitple_slides (TestMultipleSlides.MultipleSlidesTestCase.test_mulitple_slides)

----------------------------------------------------------------------

Ran 1 test in 2.447s



OK


--

```

dzhidzhoev replied:
Also, it may be worth cherry-picking this https://github.com/llvm/llvm-project/commit/58d22f1b665ca8810c4de341426247b7b5a69548 to have build commands and outputs in the log without tracing hassle.

kendalharland replied:
Ty! I'll try that out.

kendalharland replied:
Looks like after my CMake chages,this one is passing now.  I'll remove it from the PR

kendalharland replied:
Ah, sorry I spoke too soon. On windows I am seeing this fail. The test is built without debug information because this is in it's makefile:

```
C_SOURCES := main.c
MAKE_DSYM := NO

include Makefile.rules

main.o: main.c
	$(CC) $(CFLAGS_NO_DEBUG) -c $< -o $@
```

I am curious to know how this is passing for you? Without specifying `-g` or `-gdwarf` to Clang, lldb cannot resolve the name `first` and fails for me locally. But adding either of those flags fixes the problem.

dzhidzhoev replied:
Here's the execution log on my setup https://gist.github.com/dzhidzhoev/c6af2c83c6175eb1ba83c306b0ad9a15.

Build commands are
```
F:/Users/vdzhidzhoev/temp/build-lldb-native/bin/clang.exe -O0 -m64     -c F:\Users\vdzhidzhoev\temp\llvm-project\lldb\test\API\functionalities\multiple-slides/main.c -o main.o
F:/Users/vdzhidzhoev/temp/build-lldb-native/bin/clang.exe main.o -gdwarf -O0  -m64 -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make/../../../../..//include -IF:/Users/vdzhidzhoev/temp/build-lldb-native/tools/lldb/include -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\test\API\functionalities\multiple-slides -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make -include F:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make/test_common.h -fno-limit-debug-info     -fuse-ld=lld  --driver-mode=g++ -o "a.out"
```

It seems that -gdwarf is added on this line https://github.com/llvm/llvm-project/blob/8f7910a4fc08818462e0e836db74b20290cd36c7/lldb/packages/Python/lldbsuite/test/make/Makefile.rules#L270. 

Here's objdump for main.o:
```
$ bin/llvm-objdump -t  /f/Users/vdzhidzhoev/temp/build-lldb-native/lldb-test-build.noindex/functionalities/multiple-slides/TestMultipleSlides.test_mulitple_slides/main.o  | less

F:/Users/vdzhidzhoev/temp/build-lldb-native/lldb-test-build.noindex/functionalities/multiple-slides/TestMultipleSlides.test_mulitple_slides/main.o:     file format coff-x86-64

SYMBOL TABLE:
[ 0](sec  1)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .text
AUX scnlen 0x17 nreloc 2 nlnno 0 checksum 0xdb640681 assoc 1 comdat 0
[ 2](sec  2)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .data
AUX scnlen 0x4000 nreloc 0 nlnno 0 checksum 0x8ff2bf4e assoc 2 comdat 0
[ 4](sec  3)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .bss
AUX scnlen 0x0 nreloc 0 nlnno 0 checksum 0x0 assoc 3 comdat 0
[ 6](sec  4)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .xdata
AUX scnlen 0x8 nreloc 0 nlnno 0 checksum 0x1ab96b84 assoc 4 comdat 0
[ 8](sec  5)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .pdata
AUX scnlen 0xc nreloc 3 nlnno 0 checksum 0x767e3832 assoc 5 comdat 0
[10](sec  6)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000 .llvm_addrsig
AUX scnlen 0x2 nreloc 0 nlnno 0 checksum 0xe3c301f assoc 6 comdat 0
[12](sec -1)(fl 0x00)(ty   0)(scl   3) (nx 0) 0x00000000 @feat.00
[13](sec  1)(fl 0x00)(ty  20)(scl   2) (nx 0) 0x00000000 main
[14](sec  2)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x00000000 first
[15](sec  2)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x00002000 second
[16](sec -2)(fl 0x00)(ty   0)(scl  67) (nx 1) 0x00000000 .file

```

Here's "image dump symtab" from lldb.exe for a.out https://gist.github.com/dzhidzhoev/ee3bd2566f3a95b018f40c45e34d9078.

dzhidzhoev replied:
"first" is preset in both dumps

kendalharland replied:
I see. Our command lines look identical. Can you also please share the linker command that your clang invocation is generating? I am also curious to know how you built lld. Perhaps I'm missing an earlier step.

My clang output with `-v` specified is:

```
"C:\workspace\llvm-project\build\bin\clang.exe" -O0 -m64    -c C:\workspace\llvm-project\lldb\test\API\functionalities\multiple-slides/main.c -o main.o
"C:\workspace\llvm-project\build\bin\clang.exe" main.o -gdwarf -O0 -m64 -IC:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\make/../../../../../include -IC:/workspace/llvm-project/build/tools/lldb/include -IC:\workspace\llvm-project\lldb\test\API\functionalities\multiple-slides -IC:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\make -include C:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\make/test_common.h -fno-limit-debug-info   -v  -fuse-ld=lld --driver-mode=g++ -o "a.out"
clang version 20.0.0git (https://github.com/kendalharland/llvm-project bf6f1f2f5f23857eb18e08fbef29c5ca2e553bea)
Target: x86_64-unknown-windows-msvc
Thread model: posix
InstalledDir: C:\workspace\llvm-project\build\bin
 "C:\\workspace\\llvm-project\\build\\bin\\lld-link" -out:a.out -defaultlib:libcmt -defaultlib:oldnames -nologo -debug main.o
```

kendalharland replied:
For comparison, my objdump output shows no symbol table:

```
PS C:\workspace\llvm-project> C:\workspace\llvm-project\build\bin\llvm-objdump -t C:\workspace\llvm-project\build\lldb-test-build.noindex\functionalities\multiple-slides\TestMultipleSlides.test_mulitple_slides\a.out

C:\workspace\llvm-project\build\lldb-test-build.noindex\functionalities\multiple-slides\TestMultipleSlides.test_mulitple_slides\a.out:  file format coff-x86-64

SYMBOL TABLE:
```

dzhidzhoev replied:
```
$ F:/Users/vdzhidzhoev/temp/build-lldb-native/bin/clang.exe main.o -gdwarf -O0  -m64 -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make/../../../../..//include -IF:/Users/vdzhidzhoev/temp/build-lldb-native/tools/lldb/include -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\test\API\functionalities\multiple-slides -IF:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make -include F:\Users\vdzhidzhoev\temp\llvm-project\lldb\packages\Python\lldbsuite\test\make/test_common.h -fno-limit-debug-info     -fuse-ld=lld  --driver-mode=g++ -o "a.out" -v

clang version 20.0.0git (git@gitlab.ninolab.accesssoftek.com:accesssoftek/lldb-test-scripts.git 8d8d23fec72ad6ff77e15e7627ab8cc26ac36d4c)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: F:\Users\vdzhidzhoev\temp\build-lldb-native\bin
Build config: +assertions
 "F:\\Users\\vdzhidzhoev\\temp\\build-lldb-native\\bin\\lld-link" -out:a.out -defaultlib:libcmt -defaultlib:oldnames -nologo -debug main.o
```

dzhidzhoev replied:
> For comparison, my objdump output shows no symbol table:
> 
> ```
> PS C:\workspace\llvm-project> C:\workspace\llvm-project\build\bin\llvm-objdump -t C:\workspace\llvm-project\build\lldb-test-build.noindex\functionalities\multiple-slides\TestMultipleSlides.test_mulitple_slides\a.out
> 
> C:\workspace\llvm-project\build\lldb-test-build.noindex\functionalities\multiple-slides\TestMultipleSlides.test_mulitple_slides\a.out:  file format coff-x86-64
> 
> SYMBOL TABLE:
> ```

I have the same for a.out. The dump I provided (with "first") was launched for main.o.
Try launching bin/lldb, "file a.out", "image dump symtab a.out", and there should be "first" symbol. Beware that will probably show A LOT of entries.

kendalharland replied:
My output for `image dump symtab a.out` is empty. However, if I edit https://github.com/llvm/llvm-project/blob/b66aa3bfff442a5eb67f1bfcfaa148e42e49b787/lldb/test/API/functionalities/multiple-slides/Makefile#L11 to use CFLAGS instead of CFLAGS_NO_DEBUG lldb I can see symtab output similar to yours above. Do you have happen to be working with a patched or older version of that file, or have settings in your environment that force `-gdwarf` to be passed to clang? I ask because although I am now able to see debug info, I still am wondering if there's something in my local environment that is causing different behavior. Additionally, LLDB produces no output `image lookup -s first`, which still breaks the test locally but it can resolve "first" if I instead run `ta v first`. Something is preventing it from looking up the symbols for `first` and `second` even though they are clearly present as globals

kendalharland replied:
I was able to figure this out. My CMakeCache had `-D LLVM_ENABLE_DIA_SDK="OFF"` which I must have set in a previous build. I don't recall why I set this, must have been trying to repro someone else's build and forgot. This indeed passes for me now. Thanks so much for your patience and helping me debug!

kendalharland commented:
Closing this as I've been able to bring my CMake configuration in alignment with other contributors building on Windows, and these tests are now passing locally for me. Thanks everyone for the help with debuggin!

DavidSpickett commented:
Thanks for your efforts @kendalharland. If you think any of these details are worth documenting (https://lldb.llvm.org/index.html), please do so. The page sources are in `lldb/docs`.
